File: blog-frontend/src/api/track.ts
import api from './index'

export interface Track {
  id: number
  title: string
  artist?: string
  url?: string
  cover?: string
  duration?: number
  sortOrder: number
  createdAt: string
}

export const trackApi = {
  list() { return api.get<{ code: number; data: Track[] }>('/tracks') },
  listAdmin() { return api.get<{ code: number; data: Track[] }>('/admin/tracks') },
  create(data: { title: string; artist?: string; url?: string; cover?: string; duration?: number; sortOrder?: number }) {
    return api.post<{ code: number; data: Track }>('/admin/tracks', data)
  },
  update(id: number, data: { title: string; artist?: string; url?: string; cover?: string; duration?: number; sortOrder?: number }) {
    return api.put<{ code: number; data: Track }>(`/admin/tracks/${id}`, data)
  },
  delete(id: number) { return api.delete(`/admin/tracks/${id}`) },
}

File: blog-backend/src/main/java/com/blog/entity/Track.java
package com.blog.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("t_track")
public class Track {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String title;
    private String artist;
    private String url;
    private String cover;
    private Integer duration;
    private Integer sortOrder;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
}
